x-pack/legacy/plugins/siem/public/components/events_viewer/events_viewer.tsx

```diff
@@ -139,73 +145,81 @@ export const EventsViewer = React.memo<Props>(
+
+                    // TODO: Reset eventDeletedIds/eventLoadingIds on refresh/loadmore (getUpdatedAt)
+                    return (
+                      <>
+                        <HeaderSection
+                          id={id}
+                          showInspect={showInspect}
+                          subtitle={
+                            utilityBar
+                              ? undefined
+                              : `${
+                                  i18n.SHOWING
+                                }: ${totalCountMinusDeleted.toLocaleString()} ${i18n.UNIT(
+                                  totalCountMinusDeleted
+                                )}`
+                          }
+                          title={timelineTypeContext?.title ?? i18n.EVENTS}
+                        >
+                          {headerFilterGroup}
+                        </HeaderSection>
 
-                      {utilityBar?.(totalCount)}
+                        {utilityBar?.(totalCountMinusDeleted)}
 
-                      <div
-                        data-test-subj={`events-container-loading-${loading}`}
-                        style={{ width: `${width}px` }}
-                      >
-                        <ManageTimelineContext
-                          loading={loading}
-                          width={width}
-                          type={timelineTypeContext}
+                        <div
+                          data-test-subj={`events-container-loading-${loading}`}
+                          style={{ width: `${width}px` }}
                         >
-                          <TimelineRefetch
-                            id={id}
-                            inputId="global"
-                            inspect={inspect}
+                          <ManageTimelineContext
                             loading={loading}
-                            refetch={refetch}
-                          />
+                            width={width}
+                            type={timelineTypeContext}
+                          >
+                            <TimelineRefetch
+                              id={id}
+                              inputId="global"
+                              inspect={inspect}
+                              loading={loading}
+                              refetch={refetch}
+                            />
 
-                          <StatefulBody
-                            browserFields={browserFields}
-                            data={events}
-                            id={id}
-                            isEventViewer={true}
-                            height={height}
-                            sort={sort}
-                            toggleColumn={toggleColumn}
-                          />
+                            <StatefulBody
+                              browserFields={browserFields}
+                              data={events.filter(e => !deletedEventIds.includes(e._id))}
```

Comment:
I really do think that TimelineQuery need to take care of that because that will cause re-rendering